packages/frontend/src/components/card_list.tsx

```diff
@@ -0,0 +1,77 @@
+import type { QueryOptions } from "@esp-group-one/types";
+import type { ReactNode } from "react";
+import { useEffect, useState } from "react";
+
+interface CardListProps<T extends ReactNode> {
+  pageSize: number;
+  query: unknown;
+  sort: unknown;
+  nextPage: (nextQuery: QueryOptions) => Promise<T[]>;
+}
+
+export function CardList<T extends ReactNode>({
+  pageSize,
+  query,
+  sort,
+  nextPage,
+}: CardListProps<T>) {
+  const [pageNum, setPageNum] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLastPage, setIsLastPage] = useState(false);
+  let cards: T[] = [];
```

Comment:
Am I wrong believing this needs to be a state? Or is this actually really clever in the fact that it resets on a query change?